### DISCUSSION TITLE
Pasting Very Large Schematic - screen says "Paste Schematic, remaining chunks (164)

### DISCUSSION THREAD

**Discussion by tamimd67** [maruohon/litematica#817] [General]
I am attempting to paste a very large schematic into my single player flat world.  Things were happening very rapidly, but now I see a message on the screen that no longer ticks down.  It now says says "Paste Schematic, remaining chunks (164).  I am not sure what this means . . . has it stopped?   do I need to stay in the area until it all loads?  I can't find any reference to this.
![Screenshot 2024-02-09 173618](https://github.com/maruohon/litematica/assets/39843111/b237a409-b4b5-4098-b676-bcddde15a67c)

**Comment by maruohon** [comment]
The task manager that runs all of the scheduled tasks has a check that if the tick time grows to around 50 ms/tick then it won't run any (more) tasks that tick. So if the task seemingly stops progressing when you are near the remaining chunks, then that's my first guess for the reason.

In single player pasting the paste task runs on the integrated server thread, so it would mean that the server tick time is near or above 50 ms/tick. Do you have lots of entities or other potential lag sources in loaded chunks? Or you haven't forgotten the random tick speed game rule to a really high value? Also if you paste stuff really high up in the world then light updates might cause significant lag, but it probably shouldn't completely stop the process for longer periods.

**Reply by tamimd67** [reply]
I apologize, but most of that went waaaay over my head.

**Reply by maruohon** [reply]
Look on the F3 screen. (Or if you have MiniHUD then you can enable the `infoServerTPS` info line.)
![image](https://github.com/maruohon/litematica/assets/2331050/97a03879-5f35-405e-8a92-9e2f9618f0a2)

If that tick time gets near 50, then your integrated server is basically lagging and the paste task will refuse to run. And then you need to figure out and remove the lag sources.

I'll also need to add an option in some future mod versions for the past task to always run at least one operation each tick, so that you can force it to do stuff even if the game is lagging.

**Reply by tamimd67** [reply]
hmm my computer says it is running between 3 and 5.  No worries, may have to break it down to make it work.
![Screenshot 2024-02-09 183205](https://github.com/maruohon/litematica/assets/39843111/8f01f189-1f4a-41ab-9b53-a83b782b75c6)

**Reply by maruohon** [reply]
Okay well then it's not an issue with the tick times. Then it could be that Litematica somehow doesn't see the chunks as being loaded, which *could be* a mod conflict with something.
